modules/ethereum/src/verification.rs

```diff
@@ -554,6 +574,27 @@ mod tests {
 		assert_ne!(default_verify(&header), Err(Error::TimestampOverflow));
 	}
 
+	#[test]
+	fn verifies_chain_time() {
+		// header is behind
+		let header = HeaderBuilder::with_parent(&genesis())
+			.timestamp(i32::min_value() as u64)
+			.sign_by(&validator(1));
+		assert_ne!(chain_time_verify(&header), Err(Error::HeaderTimestampIsAhead));
```

Comment:
I think the test would be more valuable if it did `assert_eq` on a concrete expected `Ok/Err` value. Now it might be passing, but for instance just because the header itself is completely malformed.

Reply:
Good point!

Reply:
Done